src/seaport.ts

```diff
@@ -136,15 +140,23 @@ export class Seaport {
   }
 
   private _getSigner(accountAddress?: string): Signer {
-    if (this.signer) {
-      return this.signer;
-    }
-
     if (!(this.provider instanceof providers.JsonRpcProvider)) {
       throw new Error("Either signer or a JsonRpcProvider must be provided");
```

Comment:
This error message is wrong now. Are you sure about moving the short-circuit case above? It seems intentional by them. What's the reason to move it?

Reply:
After our convo I see that you're trying to make sure it respects `accountAddress`. Consider reconfiguring so that this error message still makes sense, and so the client still works without `this.provider` if `this.signer` is available